### PR TITLE
Chat handle creation is now an "upsert"

### LINKED ISSUE #604: Upsert chat handles
Currently, it's only really possible for a user to have one active chat handle at a time, governed by the chat adapter that Cog is currently configured to use.

In order to determine whether or not a user is updating or creating a chat handle, one approach is to add a `:show` action to the chat handles controller (currently, the `:index` action is used, which is not ideal).

However, since there can really only be one chat handle active at a time, we can simplify things by merging the current `:create` and `:update` actions into on (effectively) "upsert" operation.

To prevent confusion, we'll require the chat provider name in the API call, and return an error if that name doesn't correspond to the currently-configured chat adapter (until we can relax the "internal ID lookup logic" when chat handles are created).

Also, the database schema currently allows a single Cog user to have multiple chat handles from the same provider. That's not intentional, and can also be fixed in the course of this work.

### PR DESCRIPTION
**Pull request by christophermaier**
Previously it was a bit awkward to know whether or not you needed to
create or update a chat handle for a user. It turns out we can simplify
things, because a user can only ever have a single chat handle for a
given chat provider. Whether the user already has a chat handle or not,
we can just say we're "setting" the chat handle and basically do an
"upsert" operation on the backend.

This removes the `PUT` operation on the chat handle controller. Now, to
both create and update a user's chat handle, send a `POST` to
`/v1/users/:id/chat_handles`.

This also puts a guard in against the situation where a user tries to
set a chat handle for a provider that differs from the
currently-configured Cog chat adapter. Previously, this would crash Cog;
now it's just an error tuple.

This also pulls this chat handle logic out of the controller and into a
chat handle "repository", in keeping with the new work to consolidate
domain functionality in the core of the system.

Fixes #604

### CODE REVIEW

**Comment by jsteiner** [comment]
👍  good stuff

**Comment by kevsmith** [comment]
LGTM

File: lib/cog/repository/chat_handles.ex [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  # consult the chat provider to obtain this information.
+  #
+  # One side effect of how things are currently structured is that we
+  # only allow handles to be created or edited if the currently
+  # running chat adapter is the one for the chat provider in
+  # question. That is, if you're running Cog with the Slack adapter,
+  # you can _only_ create or update Slack chat handles.
+  defp chat_handle_params(provider_name, handle, user_id) do
+    {:ok, adapter} = Cog.chat_adapter_module
+    if String.downcase(provider_name) == adapter.name do
```
Comment (vanstee) [inline lib/cog/repository/chat_handles.ex]: This is super minor, but it's kinda a bummer that we have to `downcase` the provider_name twice. Might be worth throwing it in a variable real quick.

Comment (christophermaier) [inline lib/cog/repository/chat_handles.ex]: 😫  Aargh... one sec.

**Comment by vanstee** [comment]
Looks good. Curious about Ecto's upsert support but this should be fine.

**Pull request merged**